### PR TITLE
New year tidy

### PR DESCRIPTION
**Pull request by lawrenceakka**
Tidying up some odd and ends.

- remove code which has been deprecated for more than three releases

- move discover to its own module.  Note that `from soco import discover` will work just as it did before.

- slightly saner debug logging (clarified a few messages, removed pointless ones etc)

### CODE REVIEW

File: soco/discovery.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+""" The core module contains the SoCo class that implements
```
Comment (KennethNielsen) [inline soco/discovery.py]: This one should probably be updated

**Comment by KennethNielsen** [comment]
Reviewed with a single inline comment. I have not tested it, but it seems fairly simple, so besides the small comment +1

**Comment by lawrenceakka** [comment]
I will commit this shortly because there are a few other things waiting for it (including some bug fixes). Let me know if any last minute comments in addition to @KennethNielsen's which I have fixed.

**Comment by stefankoegl** [comment]
Nice! +1

**Pull request merged**